docs/source/devguide/tests.rst

```diff
@@ -59,6 +59,29 @@ If you want to view testing output on failure run::
 
     ctest --output-on-failure
 
+Possible Reasons for Test Failures
+----------------------------------
+
+You may find that when you run the test suite, not everything passes. First,
+make sure you have satisfied all the prerequisites above. After you have done
+that, consider the following:
+
+- When building OpenMC, make sure you run CMake with
+  ``-DCMAKE_BUILD_TYPE=Debug``. Building with a release build will result in
+  some test failures due to differences in which compiler optimizations are
+  used.
+- Because tallies involve the sum of many floating point numbers, the
+  non-associativity of floating point numbers can result in different answers
+  especially when the number of threads is high (different order of operations).
+  Thus, if you are running on a CPU with many cores, you may need to limit the
+  number of OpenMP threads used. It is recommended to set the
+  :envvar:`OMP_NUM_THREADS` environment variable to 2.
+- Recent versions of NumPy use instruction dispatch that may generate different
+  results depending the particular ISA that you are running on. To avoid issues,
+  you may need to disable AVX512 instructions. This can be done by settings the
+  :envvar:`NPY_DISABLE_CPU_FEATURES` environment variable to "AVX512F
```

Comment:
I think setting `OPENBLAS_NUM_THREADS = 1` may also be necessary for some numpy and scipy related functions to give the same answers (if they are built against libopenblas.so that is), but I'm not 100% positive on that. I'm also not familiar with `RDMAV_FORK_SAFE=1` but noticed it was also set in CI.

Reply:
IIRC the `RDMAV_FORK_SAFE` doesn't have to do with failing tests -- moreso getting MPI stuff working properly in CI